File: CMOWebApi.Models/VCModel/VCParticipantModel.cs
using CMOWebApi.Models.GeneralModel;
using System;
using System.Collections.Generic;

namespace CMOWebApi.Models.VCModel
{

    public class VCParticipantModel
    {
        public long Id { get; set; }
        public Nullable<long> Code { get; set; }
        public Nullable<long> VCCreateCode { get; set; }
        public Nullable<long> ParticipantCategoryCode { get; set; }
        public Nullable<long> ParticipantCode { get; set; }
        public string Name { get; set; }
        public string Designation { get; set; }
        public Nullable<int> DistrictCode { get; set; }
        public Nullable<long> LocationCode { get; set; }
        public string LocationTextBox { get; set; }
        public Nullable<long> ModeCode { get; set; }
        public Nullable<int> DisplayOrder { get; set; }
        public Nullable<System.DateTime> CreateDate { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public Nullable<bool> IsActive { get; set; } = true;
        public Nullable<bool> IsDeleted { get; set; } = false;
        public List<string> ParticipantCodeList { get; set; }
		public string MobileNo { get; set; }
        public Nullable<bool> IsPresent { get; set; } = true;
    }

    public class VCParticipantViewModel
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string LocationTextBox { get; set; }
        public string Designation { get; set; }
        public Nullable<bool> IsActive { get; set; }
        public string DistrictTitle { get; set; }
        public string ModeName { get; set; }
        public string LocationName { get; set; }
        public string ParticipantCategoryEnglish { get; set; }
        public string VcCreationTitle { get; set; }
        public Nullable<bool> IsPresent { get; set; }
    }

	public class VCParticipantReportViewModel
	{
        public long Id { get; set; }
        public Nullable<long> VCCreateCode { get; set; }
        public string Title { get; set; }
        public Nullable<System.DateTime> Date { get; set; }
        public string Time { get; set; }
        public string ShortDescription { get; set; }
        public Nullable<System.DateTime> VcCreationDate { get; set; }
        public Nullable<long> ChairPersonCategoryCode { get; set; }
        public string chairpersonCategoryName { get; set; }
        public Nullable<long> ChairPersonCode { get; set; }
        public string chairpersonName { get; set; }
        public Nullable<long> TypeCode { get; set; }
        public string TypeName { get; set; }
        public string StartTime { get; set; }
        public string EndTime { get; set; }
        public Nullable<decimal> NoOfHours { get; set; }
        public Nullable<decimal> NoOfMinutes { get; set; }
        public string Designation { get; set; }
        public string participantName { get; set; }
        public Nullable<System.DateTime> participantCreatedDate { get; set; }
        public string locationName { get; set; }
        public Nullable<long> locationCode { get; set; }
        public Nullable<int> DistrictCode { get; set; }
        public string DistrictTitle { get; set; }
        public Nullable<long> ModeCode { get; set; }
        public string ModeName { get; set; }
        public string ParticipantCategoryName { get; set; }
        public string MobileNo { get; set; }
        public Nullable<bool> IsPresent { get; set; }
    }

		public class VCCreationDDLModel
    {
		public long Id { get; set; }
		public Nullable<long> Code { get; set; }
		public string Title { get; set; }
		public Nullable<System.DateTime> Date { get; set; }
		public string Time { get; set; }
		public string ShortDescription { get; set; }
		public Nullable<System.DateTime> CreateDate { get; set; }
		public Nullable<System.DateTime> ModifiedDate { get; set; }
		public Nullable<int> CreatedBy { get; set; }
		public Nullable<int> ModifiedBy { get; set; }
		public Nullable<bool> IsActive { get; set; }
		public Nullable<bool> IsDelete { get; set; }
		public Nullable<long> ChairPersonCategoryCode { get; set; }
		public Nullable<long> ChairPersonCode { get; set; }
		public Nullable<long> TypeCode { get; set; }
		public string StartTime { get; set; }
		public string EndTime { get; set; }
		public Nullable<decimal> NoOfHours { get; set; }
		public Nullable<decimal> NoOfMinutes { get; set; }
		public string MeetingParticipant { get; set; }
		public string chairPersoncategoryName { get; set; }
		public string chairPersonName { get; set; }
	}

    public partial class VCParticipantDDLModel
    {
        public Nullable<long> Code { get; set; }
        public Nullable<long> ParticipantCategoryId { get; set; }
        public string DesignationHindi { get; set; }
        public string DesignationEnglish { get; set; }
        public string NameHindi { get; set; }
        public string NameEnglish { get; set; }
        public Nullable<int> DisplayOrder { get; set; }
    }

    public class VCCustomFilter
    {
		public Nullable<long> VcCode { get; set; }
		public Nullable<long> TypeCode { get; set; }
		public DateTime? Date { get; set; }
        public string VCDate { get; set; }
        public Nullable<long> ChairPersonCategoryCode { get; set; }
		public Nullable<long> ChairPersonCode { get; set; }
		public Nullable<long> DistrictCode { get; set; }
		public Nullable<long> locationCode { get; set; }
        public bool IsAllVC { get; set; }
        public int IsPresent { get; set; }
        public int IsOrderByDLPCorPCDL { get; set; }
        public long ParticipantCategoryCode { get; set; }
    }

    #region Bulk upload
    public class ParticipantTempBulkViewModel
    {
        public long Id { get; set; }
        public Nullable<long> Code { get; set; }
        public Nullable<long> VCCreateCode { get; set; }
        public Nullable<long> ParticipantCategoryCode { get; set; }
        public Nullable<long> ParticipantCode { get; set; }
        public string Name { get; set; }
        public string Designation { get; set; }
        public Nullable<int> DistrictCode { get; set; }
        public Nullable<long> LocationCode { get; set; }
        public string LocationTextBox { get; set; }
        public Nullable<long> ModeCode { get; set; }
        public Nullable<int> DisplayOrder { get; set; }
        public Nullable<System.DateTime> CreatedDate { get; set; }
        public Nullable<System.DateTime> ModifiedDate { get; set; }
        public Nullable<int> CreatedBy { get; set; }
        public Nullable<int> ModifiedBy { get; set; }
        public Nullable<bool> IsActive { get; set; }
        public Nullable<bool> IsDeleted { get; set; }
        public string MobileNo { get; set; }
        public string DistrictTitle { get; set; }
        public string NameEnglish { get; set; }
        public string DesignationEnglish { get; set; }
        public string ModeName { get; set; }
        public string LocationName { get; set; }
        public string ParticipantCategoryEnglish { get; set; }
        public string VcCreationTitle { get; set; }
    }

    public class ParticipantTempBulkModel
    {
        public Nullable<long> LocationCode { get; set; }
        public Nullable<int> DistrictCode { get; set; }
        public Nullable<long> VCCreateCode { get; set; }
    }

    #endregion
}

File: ANGULAR/ANGULAR/src/app/Shared/Model/VC/vc-participant-model.ts
import { IndexModel } from "src/app/Shared/Model/general-model";
import { DocumentUrlModel } from '../commonddl.model';
export class VCParticipantModel {
  Id: number;
  VCCreateCode: number | string;
  ParticipantCategoryCode: number | string;
  ParticipantCode: number | string;
  ParticipantCodeList: number[] | string[];
  Name: string;
  Designation: string;
  DistrictCode: number | string;
  LocationCode: number | string;
  LocationTextBox: string;
  ModeCode: number | string;
  DisplayOrder: number;
  MobileNo: string;
  constructor() {
    this.Id = 0;
    this.VCCreateCode = null;
  }
}

export class VCParticipantViewModel {
  Id: number;
  VcCreationTitle: string;
  Name: string;
  Designation: string;
  DistrictTitle: string;
  LocationName: string;
  ParticipantCategoryEnglish: string;
  IsActive: boolean;
  ModeName: string;
  IsPresent: boolean;
}

export class VCParticipantReportViewModel {
  Id: number;
  Code: number;
  Title: string;
  Date: Date;
  Time: string;
  ShortDescription: string;
  VcCreationDate: Date;
  ChairPersonCategoryCode: number;
  chairpersonCategoryName: string;
  ChairPersonCode: number;
  chairpersonName: string;
  TypeCode: number;
  TypeName: string;
  StartTime: string;
  EndTime: string;
  NoOfHours: number;
  NoOfMinutes: number;
  Designation: string;
  participantName: string;
  participantCreatedDate: Date;
  locationName: string;
  locationCode: number;
  DistrictCode: number;
  DistrictTitle: string;
  ModeCode: number;
  ModeName: string;
  IsActive: number;
  IsDelete: number;
  ParticipantCategoryCode: number;
  ParticipantCategoryName: string;
  MobileNo: string;
  IsPresent:boolean;
}

export class VCCreationDDLModel {
  Id: number;
  Code: number;
  Title: string;
  Date: Date;
  Time: string;
  ShortDescription: string;
  chairPersoncategoryName: string;
  chairPersonName: string;
}

export class VCParticipantDDLModel {
  Code: number;
  ParticipantCategoryId: number;
  DesignationHindi: string;
  DesignationEnglish: string;
  NameHindi: string;
  NameEnglish: string;
  DisplayOrder: number;
}

export class VCCustomFilter {
  VcCode: string | number;
  TypeCode: number;
  Date: Date | string;
  ChairPersonCategoryCode: number;
  ChairPersonCode: number;
  DistrictCode: number;
  LocationCode: number;
  IsAllVC: boolean;
  VCDate: string;
  IsPresent: number;
  IsOrderByDLPCorPCDL: number;
  ParticipantCategoryCode: number;

  constructor() {
    this.IsAllVC = true;
    this.IsPresent = -1;
    this.IsOrderByDLPCorPCDL = 1;
  }
}

export class ExcelFileViewModel {
  BlankDocUrl: string;
  MediaUrlList: DocumentUrlModel[] = [];
}

export class ParticipantTempBulkModel {
  LocationCode: number | string;
  DistrictCode: number | string;
  VCCreateCode: number | string;
}

//#region Bulk upload
export class ParticipantTempBulkViewModel {
  Name: string;
  Designation: string;
  MobileNo: string;
  DistrictTitle: string;
  LocationName: string;
  ParticipantCategoryEnglish: string;
  VcCreationTitle: string;
}
//#endregion
